R/network.R

```diff
@@ -0,0 +1,301 @@
+}
+
+#' Simplify a spatial network by removing multiple edges and loops.
+#'
+# nolint start
+#' Simplify the graph, removing loops and double-edge connections following
+#' [this approach](https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#simplify-network).
+#' When dropping multiple edges, keep the shortest ones.
+# nolint end
+#' @param network A network object
+#'
+#' @return A simplifed network object
+simplify_network <- function(network) {
+  network |>
+    sfnetworks::activate("edges") |>
+    dplyr::arrange(sfnetworks::edge_length()) |>
+    dplyr::filter(!tidygraph::edge_is_multiple()) |>
+    dplyr::filter(!tidygraph::edge_is_loop())
+}
+
+#' Add weights to the network.
+#'
+#' This is to prepare the network for the search of shortest paths between node
+#' pairs. The computed weights can account for edge lenghts, distance from a
+#' target geometry, and whether or not an edge falls within a specified region,
+#' which we aim to exclude from search of the shortest paths.
+#'
+#' For the i-th edge of the network, its weight \eqn{w_i} is defined in the
+#' following way:
+#' \deqn{
+#'  w_i = |e_i| + d_{geom}(e_i) + p_{buf}(e_i)
+#' }{wi = |ei| + d(geom, ei) + p(buffer, ei)}
+#' where the first term is the edge length, the second one is the distance
+#' from a target geometry (`target`, optional) and the last one is a penalty
+#' that is added if the centroid of the edge falls within a specified region
+#' (`exclude_area`, optional).
+#'
+#' Shortest paths calculated on the resulting network will thus tend to prefer
+#' edges close to `target` and to avoid edges within `exclude_area`.
+#'
+#' @param network A network object
+#' @param target Target geometry to calculate distances from, as a simple
+#'   feature geometry
+#' @param exclude_area Area that we aim to exclude from the shortest-path
+#'   search, as a simple feature geometry
+#' @param penalty Penalty (in m) that is added to the edges that falls within
+#'   the excluded area
+#' @param weight_name Name of the column in the edge table where to add the
+#'   weights
+#'
+#' @return A network object with weights added as a column in the edge table
+#' @export
+add_weights <- function(network, target = NULL, exclude_area = NULL,
+                        penalty = 1000., weight_name = "weight") {
+  edges <- sf::st_as_sf(network, "edges")
+  lengths <- sf::st_length(edges)
+
+  if (!is.null(target)) {
+    distances <- sf::st_distance(edges, target, which = "Euclidean")
+  } else {
+    distances <- 0.
+  }
+
+  if (!is.null(exclude_area)) {
+    is_inside <- edges |>
+      sf::st_centroid() |>
+      sf::st_intersects(exclude_area, sparse = FALSE) |>
+      as.numeric()
+    repellance <- units::set_units(1000., "m") * is_inside
+  } else {
+    repellance <- units::set_units(0, "m")
+  }
+
+  network |>
+    tidygraph::activate("edges") |>
+    dplyr::mutate("{weight_name}" := lengths + distances + repellance)
```

Comment:
```suggestion
    dplyr::mutate(!!rlang::sym(weight_name) := lengths + distances + repellance)
```
This version should prevent the lintr warning.

Instead of namespacing sym, rlang can also be imported in the roxygen documentation with 

```
#' @importFrom rlang !!
#' @importFrom rlang sym
```

also making sure that `rlang` is added in the DESCRIPTION file under `Imports`

Reply:
I found the following solution to work:
```r
dplyr::mutate(!!weight_name := lengths + distances + repellance)
```
with the following import in the roxygen docstring to solve the issue with ":=":
```r
#' @importFrom rlang :=
```